xdcr/replication.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2024-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+// Package xdcr implements an XDCR between different buckets. This is meant to be used for testing.
+package xdcr
+
+import "context"
+
+// Replication represents a bucket to bucket replication.
+type Replication interface {
+	// Start starts the replication.
+	Start(context.Context) error
+	// Stop terminates the replication.
+	Stop(context.Context) error
```

Comment:
Pause/Resume will be useful for our testing when we need to create conflicts.

Similarly, some way of introspecting a replication (status, stats)

Not sure if you want to wait until there's a basic Rosmar implementation before adding those - it might make it easier to abstract if there are two implementations.

Reply:
I was going to do a minimal implementation so I get something up for rosmar - I agree that there's stats needed.

I'm happy to merge as is or wait for a more complete implementation. Do you think this should go into beryllium only branch or master branch?

Reply:
`Start` and `Stop` actually can be called repeatedly so I'm not sure `Pause` and `Resume` are necessary. It depends on how this integrates with stats.

Reply:
I'm not sure what the actual difference is between pause/resume and create/delete.

I just saw it in the XDCR REST API docs and assumed we'd be following their naming/operations/semantics for this interface.

https://docs.couchbase.com/server/current/rest-api/rest-xdcr-pause-resume.html